src/uucore/src/lib/features/checksum.rs

```diff
@@ -573,109 +649,155 @@ fn identify_algo_name_and_length(
         return None;
     }
 
-    let bits = caps.name("bits").map_or(Some(None), |m| {
-        let bits_value = String::from_utf8(m.as_bytes().into())
-            .unwrap()
-            .parse::<usize>()
-            .unwrap();
-        if bits_value % 8 == 0 {
-            Some(Some(bits_value / 8))
-        } else {
-            None // Return None to signal a divisibility issue
+    let bits = if let Some(bitlen) = line_info.algo_bit_len {
+        if bitlen % 8 != 0 {
+            // The given length is wrong
+            return None;
         }
-    })?;
+        Some(bitlen / 8)
+    } else if algorithm == ALGORITHM_OPTIONS_BLAKE2B {
+        // Default length with BLAKE2b,
+        Some(64)
+    } else {
+        None
+    };
 
     Some((algorithm, bits))
 }
 
+/// Given a filename and an algorithm, compute the digest and compare it with
+/// the expected one.
+fn compute_and_check_digest_from_file(
+    filename: &[u8],
+    expected_checksum: &str,
+    mut algo: HashAlgorithm,
+    opts: ChecksumOptions,
+) -> Result<(), LineCheckError> {
+    let (filename_to_check_unescaped, prefix) = unescape_filename(filename);
+    let real_filename_to_check = os_str_from_bytes(&filename_to_check_unescaped)?;
```

Comment:
I suggest to rename `filename_to_check_unescaped` to `unescaped_filename_to_check` to match the naming of `real_filename_to_check`. And I think the `_to_check` suffices can be dropped in both cases.